src/metldata/builtin_transformations/transform_content/data_transform.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 - 2025 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data transformation logic for the `transform content` transformation"""
+
+from collections.abc import Mapping
+
+import yaml
+from jinja2 import StrictUndefined
+from jinja2.sandbox import ImmutableSandboxedEnvironment
+from schemapack import denormalize, isolate_resource
+from schemapack.spec.datapack import DataPack
+from schemapack.spec.schemapack import SchemaPack
+
+from metldata.builtin_transformations.common.utils import data_to_dict
+from metldata.builtin_transformations.transform_content.config import (
+    TransformContentConfig,
+)
+from metldata.transform.exceptions import EvitableTransformationError
+
+# configure with StrictUndefined so invalid property/dict access produces errors
+# instead of silently inserting a none value
+env = ImmutableSandboxedEnvironment(undefined=StrictUndefined)
+
+# object is too broad for what _denormalization_workaround will accept, define
+# something that's closer to the intention of JsonObjectCompatible
+ResourceContent = list | str | tuple | Mapping
+
+
+def transform_data_content(
+    *,
+    data: DataPack,
+    schemapack: SchemaPack,
+    transformation_config: TransformContentConfig,
```

Comment:
I think we should decouple the core of the transformation from the config and instead pass the two arguments here

Reply:
👍 The original ones don't.

Reply:
The original version passes the list of `instructions_by_class` including a list of instructions to the transformation functions because the transformation config can specify multiple instances of the same operation on a class, or the same operation applied to different classes with different parameters.

The new config contains arguments for a single operation on a single class. This makes it equivalent to a single instruction. That is why in this version, it looks like the config is coupled. 

Decoupling it leads to long function signatures in the `data_transform` module for this transformation. It would also make the `merge_relations` transformation more cluttered, since it involves a relatively large configuration.